libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java

```diff
@@ -299,9 +380,10 @@ private String computeWhereClause() {
                     break;
                 }
             case match:
-                String joinClause = computeSoupFtsReference() + "." + SmartStore.DOCID_COL + " = " + computeFieldReference(SmartStore.SOUP_ENTRY_ID);
-                String matchCause = field + " MATCH '" + matchKey + "' "; // statement arg binding doesn't seem to work so inlining matchKey
-                pred = joinClause + " AND " + matchCause;
+                pred = computeFieldReference(SmartStore.SOUP_ENTRY_ID) + " IN ("
```

Comment:
See https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1152/commits/99e04f5a73fe7619eaa8f3ff16208b418bfe122b for more information